docs/access/proxy.md

```diff
@@ -0,0 +1,156 @@
+# Setting up a proxy server
+
+As a first step towards a production-ready CernVM-FS setup
+we will install a [Squid forward proxy server](http://www.squid-cache.org),
+which is strongly recommended in the context of HPC systems.
+
+The proxy server will (often dramatically) **reduce the latency** for client systems,
+and hence **significantly improve start-up performance** of software provided via a CernVM-FS
+repository. In addition, it reduces the load on the Stratum 1 replica servers that
+support the CernVM-FS repositories being used.
+
+This is particularly important when running large-scale [MPI](https://en.wikipedia.org/wiki/Message_Passing_Interface)
+software, since starting the software requires that the corresponding binary and the libraries
+it depends on are available on *all* worker nodes being employed.
+
+## General recommendations
+
+It is strongly recommended to have **at least two proxy servers** available,
+to have some redundancy available in case of unexpected problems or when performance maintenance.
+
+As a rule of thumb, it is recommended to have (at least) *one proxy server for every couple
+of hundred worker nodes* (100-500).
+
+The load on the proxy servers is highly dependent on the workload mix on the client systems.
+
+## Proxy server setup
+
+### Installation
+
+First, install the `squid` package using your OS package manager:
+
+=== "For RHEL-based Linux distros (incl. CentOS, Rocky, Fedora, ...)"
+
+    ``` { .bash .copy }
+    sudo yum install -y squid
+    ```
+
+=== "For Debian-based Linux distros (incl. Ubuntu)"
+
+    ``` { .bash .copy }
+    sudo apt install -y squid
+    ```
+
+### Configuration
+
+Create a configuration file for the Squid proxy in `/etc/squid/squid.conf`.
+
+You can use the following template for this:
+
+```{ .apache .copy }
+# List of local IP addresses (separate IPs and/or CIDR notation) allowed to access your local proxy
```

Comment:
Perhaps add a comment here that this is not really necessary if you have other firewall rules that block access to the proxy.

Reply:
Should be done in a follow-up PR imho, since it's a tiny detail